File: scratch/trace_block.py
import sys
lines = open('api/index.js').readlines()
balance = 0
in_report = False
for i, line in enumerate(lines):
    if "else if (type === 'end-of-call-report') {" in line:
        in_report = True
        balance = 1
        continue
    if not in_report: continue
    diff = line.count('{') - line.count('}')
    balance += diff
    if i+1 == 3721:
        print(f"Balance at 3721 is: {balance}")
        break

File: scratch/trace_block.js
const fs = require('fs');
const lines = fs.readFileSync('api/index.js', 'utf8').split('\n');

let balance = 0;
let inReport = false;
for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    if (line.includes("else if (type === 'end-of-call-report') {")) {
        inReport = true;
        balance = 1;
        console.log(`Started at line ${i+1}`);
        continue;
    }
    if (!inReport) continue;
    
    balance += (line.match(/\{/g) || []).length;
    balance -= (line.match(/\}/g) || []).length;
    
    if (balance === 0) {
        console.log(`Block ended at line ${i+1}: ${line.trim()}`);
        break;
    }
}
